### PR TITLE
spatial overlaps 

### PR DESCRIPTION
**Pull request by jread-usgs**
@berdaniera added this for #17 . Doesn't include Alaska

### CODE REVIEW

File: R/extract_spatial.R [hunk 1]
```diff
@@ -57,28 +57,67 @@ feature_count.Spatial <- function(sp){
...
+  feature.states <- lapply(sort(state.names), function(x) list('state-name'=x, 'state-abbr' = dataRetrieval::stateCdLookup(x)))
+  return(list(states = feature.states))
 }
 
-feature_states.SpatialPolygons <- function(sp){
 
+#' @importFrom maps map
+#' @importFrom maptools map2SpatialPolygons
+#' @importFrom sp CRS
+get_states <- function(){
```
Comment (berdaniera) [inline R/extract_spatial.R]: I edited this inline, how do you usually manage changes like that?

**Comment by berdaniera** [comment]
Ok, right, still learning... I have a fork. To get the latest changes on the master I do `git pull upstream master` right? 

What is the approach for seeing and modifying these changes? Do I commit and then pull and edit?

**Comment by jread-usgs** [comment]
:+1:

If you have the upstream set up, `git pull upstream master` is my preferred way of getting changes back to your local computer. Then `git push` will push those changes to your github fork (cloud fork).

if you need to set up the upstream remote, it would look something like this:
`git remote add upstream git@github.com:USGS-R/meddle.git`

if you have git set up w/ R studio, you will see the diffs with your changes, and you can/should add commit messages to those changes:
image

Are you referring to local (your) changes, or the changes I have made here? Note that you won't be able to pull mine down w/ `git pull upstream master` until they are merged. Before that, if you want to get them locally, you can do `git pull https://github.com/jread-usgs/meddle.git master` but I would suggest avoiding that until you work through the process of creating a branch (don't mess with that for now). Just my $0.02

File: R/extract_spatial.R [hunk 2]
```diff
@@ -97,8 +96,10 @@ feature_states.Spatial <- function(sp){
 #' @importFrom maptools map2SpatialPolygons
 #' @importFrom sp CRS
 get_states <- function(){
-  usa <- map("state", fill = TRUE, plot = FALSE)
-  IDs <- sapply(strsplit(usa$names, ":"), function(x) x[1])
```
Comment (berdaniera) [inline R/extract_spatial.R]: Yep, a good reason not to do what I did!

Comment (jread-usgs) [inline R/extract_spatial.R]: ok, I guess it is trickier than that. I don't think it is possible to combine the different maps in the way that you have coded it here @berdaniera

Comment (berdaniera) [inline R/extract_spatial.R]: oh, yikes, working to fix it now... will add it to my standing pull request.

**Pull request merged**